lib/write.ml

```diff
@@ -436,33 +430,31 @@ let to_outbuf ?(std = false) ob x =
 let to_string ?buf ?(len = 256) ?std x =
   let ob =
     match buf with
-        None -> Bi_outbuf.create len
+        None -> Buffer.create len
       | Some ob ->
-          Bi_outbuf.clear ob;
+          Buffer.clear ob;
           ob
   in
   to_outbuf ?std ob x;
-  let s = Bi_outbuf.contents ob in
-  Bi_outbuf.clear ob;
+  let s = Buffer.contents ob in
+  Buffer.clear ob;
   s
 
-let to_channel ?buf ?len ?std oc x =
+let to_channel ?buf ?(len=4096) ?std oc x =
   let ob =
     match buf with
-        None -> Bi_outbuf.create_channel_writer ?len oc
+        None -> Buffer.create len
```

Comment:
This breaks `to_channel` and `to_file` since we just write to a `Buffer` and nothing else ever happens.

We need to fix this obviously but my guess is that depending on how we do it this might have an impact on performances. Just calling `Buffer.output_buffer` at the end will probably perform pretty bad comapred to the old implementation.

The fact that this can pass the CI is alarming, tomorrow I will:
1. Add basic unit tests to make sure the entry points of the API work as expected
2. Add IO versions to the benchmarks